2017/proposals/ranasinghe-kanchana.md

```diff
@@ -0,0 +1,104 @@
+This work hopes to tackle this issue to a certain extent. The Categorical Color Project will include two main tasks: building the heat-map and other scalarMappables API, and normalization and color-map APIs. Matplotlib currently has these frameworks set-up with functionality for numerical data. This functionality will be extended to support categorical data. The work of this project will be done solely using Python. Libraries used will include Numpy and Matplotlib. There will be no new (unused in Matplotlib previously) third-party libraries used in the course of this project. 
+The work of this project will stem from existing development done categorical color support in Matplotlib issues [6889](https://github.com/matplotlib/matplotlib/pull/6889) and [7383](https://github.com/matplotlib/matplotlib/issues/7383). The starting point for the project will be extending the imshow function (matplotlib.axes._axes.imshow) to support categorical data. This mainly focuses on providing it unit knowledge with regards to non-numerical data. The approach followed here will be the initial basis for this extended support for categorical data. It will be used as a foundation to build the heat-maps API, and normalization and color-map APIs. 
+Currently, the process of using matplotlib.pyplot for heat-maps and other scalarMappables includes the creation of a color-map followed by normalization of all available data to fit this color-map. Color-maps plotting functionality is threefold: sequential, divergent and qualitative. Considering the intrinsic discreteness of categorical data, the extension of the color-maps API will focus on its qualitative plotting ability. However the fundamental behavior of the color-maps API will require changes for handling categorical data as it is currently dependent on the presence of numerical scalar values in the data to be plotted. 
+With regards to the normalization function, since this extension will require working with categorical data, the concept of normalization of data seems to be rendered useless. However, the possibility of integrating this into the APIs to be built will be evaluated. Also during the course of the project focus will be given to the currently available discrete normalization methods (matplotlib.colors.BoundaryNorm) which handle data in a different sense. 
```

Comment:
Not in a matplotlib sense, where normalization is just how data gets mapped to colors....